core/src/main/java/org/apache/iceberg/Partitioning.java

```diff
@@ -298,4 +324,33 @@ private static boolean compatibleTransforms(Transform<?, ?> t1, Transform<?, ?>
         || t1.equals(Transforms.alwaysNull())
         || t2.equals(Transforms.alwaysNull());
   }
+
+  private static Set<Integer> commonPartitionFieldIds(Collection<PartitionSpec> specs) {
+    Set<Integer> commonFieldIds = Sets.newHashSet();
+
+    int specIndex = 0;
+    for (PartitionSpec spec : specs) {
+      if (specIndex == 0) {
+        commonFieldIds.addAll(activeFieldIds(spec));
+      } else {
+        commonFieldIds.retainAll(activeFieldIds(spec));
+      }
+
+      specIndex++;
+    }
+
+    ValidationException.check(
+        commonFieldIds.size() > 0 || specs.size() == 1,
+        "Specs do not have any common fields: %s",
+        specs);
+
+    return commonFieldIds;
+  }
+
+  private static List<Integer> activeFieldIds(PartitionSpec spec) {
+    return spec.fields().stream()
```

Comment:
This weeds out fields that have the same ID but were deleted in V1 tables correct?

Reply:
Yep, correct.